src/app/pages/signing/image-modal/image-modal.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, ViewChild, OnInit, ElementRef } from '@angular/core';
+import { BsModalRef, ModalDirective } from 'ngx-bootstrap/modal';
+import { ErrorAlertComponent } from '../error-alert/error-alert.component';
+import { SigningService } from '../../../core/signing.service';
+
+@Component({
+  selector: 'app-image-modal',
+  templateUrl: './image-modal.component.html',
+  styleUrls: ['./image-modal.component.css'],
+})
+export class ImageModalComponent implements OnInit {
+  static maxFileSize = 8 * 1024 * 1024; // 8 MiB in bytes
+
+  static acceptedFiles = [
+    '.png',
+    '.jpg',
+    '.jpeg',
+    '.tif',
+    '.tiff',
+    '.bmp',
+    '.gif',
+    '.jfif',
+    '.pjp',
+  ];
```

Comment:
Instead of hard coding all the supported file types, couldn't we just use the MIME type? Could make the application a bit more flexible. For example, there is still the `.jpe` extension (among others) missing for JPEG https://en.wikipedia.org/wiki/JPEG.

Reply:
This is a good point, but the reason for switching to this instead of the original "image/*" is because we don't support every single image format. For example, `.svg` doesn't work.

Reply:
This was actually to prevent uploads using the 'All Files' option in the selection popup, which we cannot do anything about natively in html. This would enforce the supported filetypes much better as it would check in the back before doing anything.

Reply:
@jonapap see [d869c4f](https://github.com/virtrolio/virtrolio-site/pull/422/commits/d869c4f7d7547145f53bf737aaa71ae3556205b6)